test/XGreeter.t.sol

```diff
@@ -38,6 +38,7 @@ contract XGreeterTest is Test {
 
     // @dev Test that xgreet reverts if the fee is insufficient
     function test_xgreet_insufficientFee_reverts() public {
+        vm.deal(address(xgreeter), 1 ether); // reverts with no reason if contract has no balance
```

Comment:
It should revert with "XApp: insufficient funds" (see  [here](https://github.com/omni-network/omni/blob/main/contracts/src/pkg/XApp.sol#L62))

You may have to update omni lib (I see you did that but maybe not to right commit)